tests/rados/rados_prep.py

```diff
@@ -369,3 +497,43 @@ def create_pool(node: CephAdmin, pool_name: str, pg_num: int = 64, **kwargs) ->
 
     log.info(f"Created pool {pool_name} successfully")
     return True
+
+
+def run_ceph_command(node: CephAdmin, action: str) -> dict:
+    """
+    Runs ceph commands with json tag for the action specified otherwise treats action as command
+    and returns formatted output
+    Args:
+        node: Cephadm node where the commands need to be executed
+        action: The various commands that can be run are:
+            1. "ceph_report" -> cluster status
+            2. "daemon_info" -> info on all daemons running on cluster
+            3. "mgr_modules" -> list of modules available in mgr module
+            4. "cluster_usage" -> provides details about cluster/pool usage
+            5. "config_dump" -> Config changes made on cluster ( any daemon )
+            6. "cluster_usage_detail" -> detailed clustr usage info including limits set
+            7. "osd_details" -> info on OSD collected from Crush tree
+            8. "mon_details" -> monitor quorum and epoch details
+
+    Returns:
+    """
+
+    cmd_map = {
+        "ceph_report": r"ceph report",
+        "daemon_info": r"ceph orch ps",
+        "mgr_modules": r"ceph mgr module ls",
+        "cluster_usage": r"ceph df",
+        "config_dump": r"ceph config dump",
+        "cluster_usage_detail": r"ceph df detail",
+        "osd_details": r"ceph osd df tree",
+        "mon_details": r"ceph mon dump",
+    }
+    # If command is not found in cmdmap, considering action string sent as the command
+    if not cmd_map.get(action):
+        cmd = f"{action} -f json-pretty"
```

Comment:
Any particular reason why `json-pretty` is used. Switching to `json` can help avoid different newline and indentation characters creeping into the output.

Reply:
changed to json